salalib/axialmap.cpp

```diff
@@ -1363,20 +1366,23 @@ int ShapeGraphs::convertDataToAxial(Communicator *comm, const std::string& name,
           for (size_t k = 1; output.getColumnIndex(colname) != -1; k++){
              colname = dXstring::formatString((int)k,input.getColumnName(i) + " %d");
           }
-          columns.push_back( output.insertColumn(colname));
+          attrCols.push_back( output.insertColumn(colname));
        }
    }
 
-   for (size_t k = 0; k < lines.size(); k++) {
-      if (copydata){
-          AttributeTable& input = shapemap.getAttributeTable();
-          for ( int i = 0; i < input.getColumnCount(); ++i)
-          {
-              extraAttributes[columns[i]] = input.getValue(k, i);
-          }
-      }
-      usermap.makeLineShapeWithRef(lines[k], keys[k], false, false, extraAttributes);
-   }
+    AttributeTable& input = shapemap.getAttributeTable();
+    auto keyIter = keys.begin();
+    for (auto& line: lines) {
+        if (copydata){
+            int rowid = input.getRowid(keyIter->second);

```

Comment:
Are you sure this is correct? I am pretty sure that getValue() takes a row index, not a row id... Hmm, maybe not. I hate this data structure.

Reply:
It seems to me that rowid is the row index (from attributes.h:263):
```
int getRowid(const int key) const
      { size_t i = searchindex(key); return (i == paftl::npos) ? -1 : int(i);}
```